contributing/giving-constructive-criticism.rst

```diff
@@ -0,0 +1,155 @@
+help you. Take a deep breath, count to 10 and try to *clearly* explain what problems
+you encounter.
+
+Gender pronouns
+---------------
+
+While not "formally" required it's better to use gender-neutral pronouns.
+Unless someone "indicated" their pronouns, use "they", "them" instead of
+"he", "she", "his", "hers", "his/hers", "he/she", etc.
+
+And instead of saying "guys" it's better to use "folks", or Symfonions
+if you like that.
+
+Giving positive feedback
+------------------------
+
+While reviewing issues and pull request you may run into some suggestions
+(including patches) that don't reflect your ideas, are not good, or downright wrong.
+
+Now, instead of writing a lengthy comment why you think they are wrong,
+consider the amount of work and time the author has spent on their idea
+and how you're response would make them feel;
+
+* Are you making a point only because you believe they are wrong?
+
+* Did you correctly understand their intention? or are you making assumptions?
+
+Maybe you can ask for clarification? suggest an alternative? or
+provide a simple explanation *why* disagree with their proposal.
+
+Even if something *is* really wrong or "a bad idea", stay respectful and
+don't get into endless you-are-wrong discussions or "flame wars".
+
+* Good questions avoid judgment and avoid assumptions about the author's
+  perspective.
+
+* Avoid using terms that could be seen as referring to personal traits. ("dumb",
+  "stupid"). Assume everyone is intelligent and well-meaning.
+
+* Be explicit. Remember people don't always understand you're intentions online.
+
+* Don't use hyperbole. ("always", "never", "endlessly", "nothing", "worst",
+  "horrible", "terrible").
+
+Instead off saying "This is a horrible idea" use something like
+"This can introduce issues in the future. Please make sure this is safe to use."
+
+Asking for changes
+------------------
+
+Rarely something is perfect from the start, while the code itself is good.
+It may not be optimal or conform the Symfony coding style.
+
+Again, understand the author already spent time on the issue and asking
+for (small) changes may be misinterpreted or seen as a personal attack.
+
+Be thankful for their work (so far), stay positive and really help them
+to make the contribution a great one. *Especially if they are a first
+time contributor.*
+
+Use words like "Please", "Thank you" and "Could you" instead of making demands;
+
+* "Thank you for your work so far. I left some suggestions for improvement
+  to make the code more readable."
+
+* "Your code contains some coding-style problems, can you fix these before
+  we merge? Thank you"
+
+* "Please use 4 spaces instead of tabs", "This needs be on the previous line";
+
+  During a pull request review you can usually leave more then one comment,
+  you don't have to use "Please" all the time. But it wouldn't hurt.
+
+It may not seem like much, but saying "Thank you" does make others feel
+more welcome.
+
+Making jokes (using humor)
```

Comment:
After some though about this, I'd say we should ask to NOT use humor ever. Humor is very prone to misunderstanding, specially when in a huge community as Symfony, where you have lots of different cultures and most people are not English native.

I'd recommend to not use humor when reviewing things and leave that for Symfony's Slack, where real-time communication with people you know well provides the perfect conditions for informal and humorous conversations.